Mathlib/Data/Nat/Log.lean

```diff
@@ -197,6 +206,18 @@ theorem add_pred_div_lt {b n : ℕ} (hb : 1 < b) (hn : 2 ≤ n) : (n + b - 1) /
     succ_pred_eq_of_pos (by omega)]
   exact Nat.add_le_mul hn hb
 
+lemma log2_eq_log_two {n : ℕ} : Nat.log2 n = Nat.log 2 n := by
+  induction n using Nat.strongRec with
+  | _ =>
+    rename_i n ih
+    unfold Nat.log2 Nat.log
+    by_cases h : n = 0
+    · simp [h]
+    · congr
+      simp only [ge_iff_le, lt_succ_self, and_true]
+      exact ih (n/2) (Nat.div_lt_self (Nat.pos_of_ne_zero h) (by simp))
```

Comment:
Here's a golf. And I wonder if the `suffices` is already in mathlib somewhere?
```suggestion
  suffices h : ∀ {m}, m ≤ Nat.log2 n ↔ m ≤ Nat.log 2 n from
    le_antisymm (h.mp le_rfl) (h.mpr le_rfl)
  rcases eq_or_ne n 0 with rfl | hn
  · simp only [log2_zero, le_zero_eq, log_zero_right, implies_true]
  simp only [Nat.le_log2 hn, ← Nat.pow_le_iff_le_log Nat.one_lt_two hn, implies_true]
```

Reply:
Suggestion committed!